src/Trig.cpp

```diff
@@ -76,8 +76,8 @@ bool NAS2D::lineIntersectsCircle(const Point_2d& p, const Point_2d& q, const Poi
 
 	float t = -(((p.x() - c.x()) * dx) + ((p.y() - c.y()) * dy)) / ((dx * dx) + (dy * dy));
 
-	if (t < 0.0) { t = 0.0; }
-	else if (t > 1.0) { t = 1.0; }
+	if (t < 0.0f) { t = 0.0; }
+	else if (t > 1.0f) { t = 1.0; }
```

Comment:
This is just `t = std::clamp(t, 0.0f, 1.0f);`

Reply:
Indeed it is.